### PR TITLE
Multilanguage: Prevent display of the edit associations view when the item is new or its language is set to All.

### PR DESCRIPTION
**Pull request by infograf768**
See https://github.com/joomla/joomla-cms/pull/10913

Before this patch, one could access the Edit Associations view through the Associations tab, even when the item was new or its language was set to "All languages".

The Association tab would show All existing Content Languages which made no sense. Example for a multilingual site with 7 languages, we would get all 7 languages instead of 6 possibilities:

screen shot 2016-06-25 at 07 59 11

Until this PR, one would only get a Message after Save or Save and Close the item (#10913)

We had 2 solutions: one was to not display the Associations tab at all when the item was new and/or when its language was set to All.
I chose to go on displaying the Associations tab to avoid confusion and just display an alert in it.

To test, install a multilang site.
Create a new item (article, category, contact, newsfeed, menu item) and before save, go to the Associations tab. 
Or edit an item which language have been set to ALL and go to the Associations tab.

One should now get

screen shot 2016-06-25 at 07 36 47

@brianteeman 
Please look at the new string value.

### CODE REVIEW

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on bf4dff4a8e500b3afad1e192f9c3f55ce014a8e1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10926">issues.joomla.org/joomla-cms/10926</a>.</sub>

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on bf4dff4a8e500b3afad1e192f9c3f55ce014a8e1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10926">issues.joomla.org/joomla-cms/10926</a>.</sub>

File: administrator/components/com_categories/views/category/tmpl/edit_associations.php [hunk 1]
```diff
@@ -9,4 +9,11 @@
...
 defined('_JEXEC') or die;
 
-echo JLayoutHelper::render('joomla.edit.associations', $this);
+if ($this->item->id != 0 && $this->item->language != '*')
+{
+	echo JLayoutHelper::render('joomla.edit.associations', $this);
+}
+else
+{
+	echo '<div class="alert alert-no-items">' . JText::_('JGLOBAL_ASSOC_NOT_POSSIBLE') . '</div>';
```
Comment (andrepereiradasilva) [inline administrator/components/com_categories/views/category/tmpl/edit_associations.php]: IMHO should be a notice, not a warning.
Also, the class shouldn't be `alert-info` (or `alert-warning` for a warning).

The same applies to all related changes.

Comment (ggppdk) [inline administrator/components/com_categories/views/category/tmpl/edit_associations.php]: Yes 'alert-no-items' should not be used in this case, it is not a search list or anything similar

**Comment by infograf768** [comment]
Done, we now get:

screen shot 2016-06-25 at 11 19 38

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on ddbaa2fdc9fe0f898ab4f6a2e863b4a106055597<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10926">issues.joomla.org/joomla-cms/10926</a>.</sub>

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on ddbaa2fdc9fe0f898ab4f6a2e863b4a106055597<br /><br />Tested all 5 forms, switching language, worked as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10926">issues.joomla.org/joomla-cms/10926</a>.</sub>

**Comment by zero-24** [comment]
RTC based on tests. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10926">issues.joomla.org/joomla-cms/10926</a>.</sub>

**Comment by infograf768** [comment]
@roland-d 
if we merge the PRs with lang changes in 3.6.1, can you please change the milestone for this to 3.6.1?

**Comment by zero-24** [comment]
bot please

**Pull request merged**